app/Helpers/MSGraphAPI/UserByApp.php

```diff
@@ -0,0 +1,216 @@
+<?php
+namespace App\Helpers\MSGraphAPI;
+
+
+use Microsoft\Graph\Graph;
+use GuzzleHttp;
+use Illuminate\Support\Str;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+
+class UserByApp {
+    protected $employeeNumberProperty ;
+    protected $descriptionProperty ;
+    protected $mailProperty ;
+    protected $RoleAdminId;
+    protected $RoleUserId;
+
+    public function __construct()
+    {
+        $extension_app_id = Str::remove('-', config('services.azure.extension_app_id'));
+        $this->employeeNumberProperty = 'extension_' . $extension_app_id . '_employeeNumber';
+        $this->descriptionProperty = 'extension_' . $extension_app_id . '_description';
+        $this->mailProperty = 'extension_' . $extension_app_id . '_mail';
+    }
+
+
+
+    public function getUsers($topN=null,$searchProperty=null,$searchValue=null,$checkAdmin=False) {
+        $token = $this->_token();
+        if ($token) {
+            if($checkAdmin){
+                $this->getApplicationRoles();
+            }
+            $graph = new Graph();
+            $graph->setApiVersion("v1.0");
+            $graph->setAccessToken($token);
+            // if request for max items then append to query
+            $topN=$topN ? '&$top=' . $topN : null;
+            // if request for extensionProperty then translate the propertyName
+            switch($searchProperty){
```

Comment:
Dit in een aparte functie zetten die dan gewoon de search string terug geeft.
Is leesbaarder dan een grote functie met van alles en nog wat